app/category/[...category]/page.tsx

```diff
@@ -0,0 +1,81 @@
+"use client";
+import { useEffect } from "react";
+import { useRouter, notFound } from "next/navigation";
+import { useAppDispatch, useAppSelector } from "@/lib/hookts";
+import {
+  setCategoryFilter,
+  fetchProducts,
+  sortProductsByCategory,
+} from "@/lib/features/products/FetchDataSlice";
+import Categories from "@/lib/enums/ProductCategories";
+import { RootState } from "@/lib/store";
+import ProductCard from "@/components/displayProducts/ProductCard";
+import InfoModal from "@/components/infoModal/InfoModal";
+import Link from "next/link";
+import { BsFillHouseDownFill } from "react-icons/bs";
+import { TbMathGreater } from "react-icons/tb";
+import { AnimatePresence } from "framer-motion";
+import ClipLoader from "react-spinners/ClipLoader";
+
+const CategoryPage = ({ params }: { params: { category: string[] } }) => {
+  const dispatch = useAppDispatch();
+  const { filteredProducts, categoryFilter, loading } = useAppSelector(
+    (state: RootState) => state.products
+  );
+
+  useEffect(() => {
+    const category = params.category[0];
+    if (
+      category &&
```

Comment:
@Kamildeeal create a separate hook for this logic like `useCategoryPage`. Also try to avoid big blocks of code in `useEffect`. Better create a method that will be only triggered in useEffect